0000-pre-shared-key-v2.md

```diff
@@ -122,74 +137,9 @@ encryption_key = hmac_sha256(shared_secret, "ilp_psk_encryption")
-\                                                               \
-\                                                               \
-|                          Junk Data                            |
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-```
-
-| Field | Type | Description |
-|---|---|---|
-| Type (T) | 1-Byte UInt (UInt8) | Always `0` to indicate the packet is a quote request |
-| Quote ID | 16-Byte UInt | Unique identifier for this quote |
-| Extensions (E) | 1-Byte UInt (UInt8) | Always `0`. Reserved for future use to indicate the presence of extension fields |
-| Junk Data | 0-32737 Bytes | _(Optional)_ Extra data included to obscure what the ILP payment is for. Receivers MUST ignore this data. Senders SHOULD include either the maximum or a random number of empty (0) bytes |
-
-**TODO:** should the quote request / response carry application data?
-
-#### Quote Response
-
-The following is encrypted, as defined [above](#encryption), and set as the data field of an ILP Error sent by the receiver when they reject the incoming quote request.
-
-The receiver SHOULD set the ILP Error Code to `F99: Application Error`.
-
-```
-(Numbers represent bytes)
+### Data Packet
 
- 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|T|            Quote ID           | Amount Rcved  |E| Junk Data |
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-|                          Junk Data                            |
-\                                                               \
-\                                                               \
-|                          Junk Data                            |
-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-```
-
-| Field | Type | Description |
-|---|---|---|
-| Type (T) | 1-Byte UInt (UInt8) | Always `1` to indicate the packet is a quote response |
-| Quote ID | 16-Byte UInt | Same unique identifier from the request. Senders SHOULD check that the id on the response matches the request to prevent replay attacks |
-| Amount Received | 8-Byte UInt (UInt64) | Amount that arrived in the incoming transfer to the receiver |
-| Extensions (E) | 1-Byte UInt (UInt8) | Always `0`. Reserved for future use to indicate the presence of extension fields |
-| Junk Data | 0-32729 Bytes | _(Optional)_ Extra data included to obscure what the ILP payment is for. Receivers MUST ignore this data. Senders SHOULD include either the maximum or a random number of empty (0) bytes |
-
-#### Quote Condition
-
-Senders SHOULD use a random 32-byte value as the payment condition.
-
-#### Quote Fulfillment
-
-Not applicable. The quote is intentionally unfulfillable.
-
-### Payment Chunks
-
-#### Payment Chunk Request
+All PSKv2.0 requests and responses encode their data in the same byte format, though the meanings of the fields differ depending on whether it is a request or a response.
```

Comment:
What's the motivation for defining a binary packet format?

Reply:
1. We need some encoding format
2. There's no reason to have the overhead of including the keys as strings, since we know what they are up front
3. Might as well make it consistent with the rest of our stack by using OER

Reply:
I think pushing OER (or any binary encoding) up to this layer is a mistake because it is passed opaquely through all intermediaries so the interoperability requirement is significantly diminished.

Only the sender and receiver need to interpret this so it's more likely to be implemented by clients than connectors and something more common at this layer (like JSON) just seems like a better natural fit.

Reply:
It is only the sender and receiver that need to implement this, but the sender and receiver also need to support OER to send and receive ILP packets.

Reply:
Agreed, I think many application layer protocols will use JSON or some other "friendlier" encoding. If that's the case though, that's all the more reason to have as minimal byte overhead as possible in the transport layer so the application layer can send more data.

In most cases I would expect the ILP and PSK2 modules to be implemented roughly at the same time, by the same people. If you're building a sender or receiver right now you'll need both, and I think that is a good place to get started for an implementation.